source/RialtoGStreamerMSEVideoSink.cpp

```diff
@@ -232,10 +243,22 @@ static void rialto_mse_video_sink_get_property(GObject *object, guint propId, GV
     case PROP_WINDOW_SET:
         if (!client)
         {
-            std::unique_lock lock{priv->rectangleMutex};
-            g_value_set_string(value, priv->videoRectangle.c_str());
+            // It is possible that within rialto_mse_video_sink_change_state() that:-
```

Comment:
This client reassigment is not very readable for me and I would do it in one go. I would remove client assignment, which is above the switch and do something like:
```
case XXX:
{
std::shared_ptr<GStreamerMSEMediaPlayerClient> client;
{
mutex.lock()
client = getMediaPlayerClient()
if (!client)
{
LOG
cache value
break;
}
}
client->XXX
}
```